## DP: Fix compilation error in drivers/input/misc/ixp4xx-beeper.c
## DP: Patch author: Alessandro Zummo <a.zummo@towertech.it>
## DP: Upstream status: submitted

Date: Tue, 14 Feb 2006 22:51:10 +0000
From: Martin Michlmayr <tbm@cyrius.com>
To: Dmitry Torokhov <dtor_core@ameritech.net>, akpm@osdl.org
Cc: Alessandro Zummo <a.zummo@towertech.it>
Subject: [PATCH] input: fix compilation error in drivers/input/misc/ixp4xx-beeper.c

From: Alessandro Zummo <a.zummo@towertech.it>

Fix compilation error of drivers/input/misc/ixp4xx-beeper.c by including
linux/interrupt.h.

  CC      drivers/input/misc/ixp4xx-beeper.o
drivers/input/misc/ixp4xx-beeper.c:81: error: syntax error before ‘ixp4xx_spkr_interrupt’
drivers/input/misc/ixp4xx-beeper.c:82: warning: return type defaults to ‘int’
drivers/input/misc/ixp4xx-beeper.c: In function ‘ixp4xx_spkr_interrupt’:
drivers/input/misc/ixp4xx-beeper.c:89: error: ‘IRQ_HANDLED’ undeclared (first use in this function)
drivers/input/misc/ixp4xx-beeper.c:89: error: (Each undeclared identifier is reported only once
drivers/input/misc/ixp4xx-beeper.c:89: error: for each function it appears in.)
drivers/input/misc/ixp4xx-beeper.c: In function ‘ixp4xx_spkr_probe’:
drivers/input/misc/ixp4xx-beeper.c:114: warning: implicit declaration of function ‘request_irq’
drivers/input/misc/ixp4xx-beeper.c:128: warning: implicit declaration of function ‘free_irq’
make[3]: *** [drivers/input/misc/ixp4xx-beeper.o] Error 1
make[2]: *** [drivers/input/misc] Error 2
make[1]: *** [drivers/input] Error 2

Signed-off-by: Alessandro Zummo <a.zummo@towertech.it>
Signed-off-by: Martin Michlmayr <tbm@cyrius.com>

---

 drivers/input/misc/ixp4xx-beeper.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-nslu2.orig/drivers/input/misc/ixp4xx-beeper.c	2006-02-06 20:37:19.000000000 +0100
+++ linux-nslu2/drivers/input/misc/ixp4xx-beeper.c	2006-02-06 22:04:59.000000000 +0100
@@ -19,6 +19,7 @@
 #include <linux/input.h>
 #include <linux/delay.h>
 #include <linux/platform_device.h>
+#include <linux/interrupt.h>
 #include <asm/hardware.h>
 
 MODULE_AUTHOR("Alessandro Zummo <a.zummo@towertech.it>");

